pkg/asset/installconfig/gcp/client.go

```diff
@@ -222,6 +225,31 @@ func (c *Client) GetProjects(ctx context.Context) (map[string]string, error) {
 	return projects, nil
 }
 
+// GetProjectRegions gets the regions that are valid for the project. An error is returned when unsuccessful
+func (c *Client) GetProjectRegions(ctx context.Context, project string) (map[string]string, error) {
+	svc, err := c.getComputeService(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	gctx := context.Background()
```

Comment:
I see from previous reviews that you were told not to reuse contexts, but is this correct? There's no way to propagate cancellation/timeout through the argument's context here since you're creating a new one. Perhaps a similar pattern to the one used in `GetZones` below would be more appropriate?
```
	ctx, cancel := context.WithTimeout(ctx, 1*time.Minute)
	defer cancel()
	svc, err := c.getComputeService(ctx)
	if err != nil {
		return nil, err
	}
        r, err := svc.Regions.List(project).Context(ctx).Do()
        ...
```

Reply:
The concern here was that the same context that was passed in as an arg is assigned again. The other functions may do this, but is it the correct approach? Reusing contexts appears to be the accepted practice on server side software. Its easy enough to change, but should I? @r4f4  @patrickdillon

Reply:
I have been discussing this with @jhixson74 here: https://github.com/openshift/installer/pull/5697#discussion_r863925907 which includes a link to go playground that illustrates the problem I was trying to describe. Note that the playground does not indicate WHAT TO DO, but  just tries to illustrate the problem as I see it. 

Calling it a problem, might be a strong word. I just want to make sure we're aware of what we're doing. @r4f4's example above gives 1 minute total to get the compute service and list the regions. Is that what we want to do? Why not give 30 seconds each? or a minute each? 

When adding a deadline/timeout to a context a child context is created from the parent context, so it should be well supported to create multiple timeouts/deadlines from a parent context. If we do want to pursue that approach I just think we need to figure out the right style/pattern. Does it work to just reuse the `ctx variable` and shadow the existing `ctx` value? This is not what the go playground does but I think it's worth checking out.

Reply:
It actually looks like AN accepted practice is to use the parent (context parameter) to create the one used in this function. I think that I should also use the parent to create the second context. 

```
gctx := context.WithTimeout(ctx, 1*time.Minute)
```

would become 

```
gctx := context.Background()
```

To avoid confusion, maybe we should change the name of the context that we create from the parent:
```
ctx, cancel := context.WithTimeout(ctx, 1*time.Minute)
```

to
```
child_ctx, cancel := context.WithTimeout(ctx, 1*time.Minute)
```

We can also change them to 30 seconds or something much more reasonable.